source/MissionAction.h

```diff
@@ -41,6 +44,11 @@ class UI;
 // In addition to performing a GameAction, a MissionAction can gate the task on
 // the ownership of specific outfits and also display dialogs or conversations.
 class MissionAction {
+public:
+	using DialogItem = std::pair<std::string, std::shared_ptr<ConditionSet>>;
+	using DialogList = std::vector<DialogItem>;
+
+
```

Comment:
These should not be public.
I'm not sure they are even necessary.
`DialogItem` only appears in these two lines.
`DialogList` appears in 6 places:
- here;
- in the name of the method `MissionAction::GetDialogList`, in both the declaration and definition;
- the return type in the declaration and definition of `MissionAction::GetDialogList`;
- the type of the field `MissionAction::dialog`.

Given that `MissionAction::GetDialogList` doesn't appear to ever be used, that only leaves this line and the line:
```c++
	DialogList dialog;
```
Not really much point having the aliases.

These lines can just be removed and the field declaration changed to:
```c++
	std::vector<std::pair<std::string, std::shared_ptr<ConditionSet>>> dialog;
```

But, I don't think there is much value to using shared pointers here.
Whatever reduction in runtime memory usage it might provide isn't worth the increased complexity of the code here.
And, in my testing, it doesn't provide a meaningful reduction in memory usage at all. Memory usage actually appears lower when I don't have it using shared pointers at all.
```c++
	std::vector<std::pair<std::string, ConditionSet>> dialog;
```

Reply:
Those types are gone. There's a new inner class to support anonymous phrases. The resulting code is cleaner, too.